### ISSUE TITLE
WITH templates collision between names of variables and functions

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#2383]
**Describe the bug**
When using `WITH` templates, users can define variables with custom names:
```
WITH(
now = sum(rate(my_metric_total)),
before = sum(rate(my_metric_total) offset 1h)
)
now/before*100
```

This expression [will fail](https://play.victoriametrics.com/promql/expand-with-exprs?query=WITH%28%0D%0Anow+%3D+sum%28rate%28my_metric_total%29%29%2C%0D%0Abefore+%3D+sum%28rate%28my_metric_total%29+offset+1h%29%0D%0A%29%0D%0Anow%2Fbefore*100&commit=Expand) starting from [v1.70](https://docs.victoriametrics.com/CHANGELOG.html#v1700) where new MetricsQL function `now()` was introduced.
But for versions below [v1.70](https://docs.victoriametrics.com/CHANGELOG.html#v1700) the expression will be valid.

**To Reproduce**
See https://play.victoriametrics.com/promql/expand-with-exprs?query=WITH%28%0D%0Anow+%3D+sum%28rate%28my_metric_total%29%29%2C%0D%0Abefore+%3D+sum%28rate%28my_metric_total%29+offset+1h%29%0D%0A%29%0D%0Anow%2Fbefore*100&commit=Expand

**Expected behavior**
Local variable names should not conflict with function names. Otherwise, adding new function to MetricsQL will always have a chance to affect existing user queries.

cc @valyala

**Comment by hagen1778** [comment]
Issue has been moved to https://github.com/VictoriaMetrics/metricsql/issues/5

**Comment by valyala** [comment]
The bugfix has been included in [VictoriaMetrics v1.76.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Now it is possible to use built-in function names as `with template` names.
